src/Microsoft.Sbom.Api/PackageDetails/ComponentDetailsUtils/MavenUtils.cs

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Xml;
+using Microsoft.ComponentDetection.Contracts.BcdeModels;
+using Microsoft.Sbom.Api.Output.Telemetry;
+using Microsoft.Sbom.Common;
+using Serilog;
+
+namespace Microsoft.Sbom.Api.PackageDetails;
+
+/// <summary>
+/// Utilities for retrieving information from maven packages that may not be present on the buildDropPath.
+/// </summary>
+public class MavenUtils : IMavenUtils
+{
+    private readonly IFileSystemUtils fileSystemUtils;
+    private readonly ILogger log;
+    private readonly IRecorder recorder;
+
+    private static readonly string EnvHomePath = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "HOMEPATH" : "HOME";
+    private static readonly string HomePath = Environment.GetEnvironmentVariable(EnvHomePath);
+    private static readonly string MavenPackagesPath = $"{HomePath}/.m2/repository";
+
+    private bool MavenPackagesPathHasReadPermissions => fileSystemUtils.DirectoryHasReadPermissions(MavenPackagesPath);
+
+    private readonly string userDefinedLocalRepositoryPath;
+
+    public MavenUtils(IFileSystemUtils fileSystemUtils, ILogger log, IRecorder recorder)
+    {
+        this.fileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
+        this.log = log ?? throw new ArgumentNullException(nameof(log));
+        this.recorder = recorder ?? throw new ArgumentNullException(nameof(recorder));
+
+        userDefinedLocalRepositoryPath = GetLocalRepositoryPath();
```

Comment:
I moved this call here in order to avoid a call to this method for every Maven component detected. If there is a better way to do this let me know 😄